test/api_integration/apis/saved_objects/find.js

```diff
@@ -165,6 +165,22 @@ export default function ({ getService }) {
               });
             })
         ));
+
+        it('KQL syntax error should return 400 with Bad Request', async () => (
+          await supertest
+            .get('/api/saved_objects/_find?type=visualization&filter=dashboard.attributes.title:foo<invalid')
```

Comment:
To really test the syntax, you should have  the `type=dashboard` if not you will get a 400 because your type is not matching with the filter